### PR TITLE
automate publishing to winget

### PR DESCRIPTION
**Pull request by vedantmgoyal2009**
This action automatically generates manifests for WinGet Community Repository ([microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs)) and submits them.

Before merging this:
1. Please add a GitHub token with `public_repo` scope as a repository secret and rename the secret name in the workflow.
2. Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under @kubernetes.

Fixes https://github.com/kubernetes/minikube/issues/14190

Edit:
Sorry for spaming the PR, but the bot is really good. It makes the review process interesting 😆

### CODE REVIEW

**Comment by vedantmgoyal2009** [comment]
> Can you fork it on your personal account?
>
> If yes, I can modify the workflow to use the fork from your personal account instead of organization account :)
>
> Note: The GitHub Personal Access Token (PAT) and the fork should be from the same account.

@spowelljr please reply so that we can move forward...

**Comment by vedantmgoyal2009** [comment]
> Can you fork it on your personal account?
>
> If yes, I can modify the workflow to use the fork from your personal account instead of organization account :)
>
> Note: The GitHub Personal Access Token (PAT) and the fork should be from the same account.

@spowelljr please reply so that we can move forward...

**Comment by vedantmgoyal2009** [comment]
> @vedantmgoyal2009
> 
> > Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under @kubernetes
> 
> That's not something we can simply do, getting new repos created under kubernetes requires a review process and in which it would likely be rejected since it's not a repo core to Kubernetes. If that's a requirement we will likely have to abandon this implementation.

@k8s-ci-robot can also be used for this purpose...

**Comment by vedantmgoyal2009** [comment]
cc @spowelljr

**Comment by vedantmgoyal2009** [comment]
Can we fork winget-pkgs repo under @k8s-ci-robot or @minikube-bot account?

If yes, I can modify the workflow to use the fork from your personal account instead of organization account :)

Note: The GitHub Personal Access Token (PAT) and the fork should be from the same account.

**Comment by vedantmgoyal2009** [comment]
> @vedantmgoyal2009
> 
> > Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under @kubernetes
> 
> That's not something we can simply do, getting new repos created under kubernetes requires a review process and in which it would likely be rejected since it's not a repo core to Kubernetes. If that's a requirement we will likely have to abandon this implementation.

Can we fork winget-pkgs repo under @k8s-ci-robot or @minikube-bot account?

If yes, I can modify the workflow to use the fork from your personal account instead of organization account :)

Note: The GitHub Personal Access Token (PAT) and the fork should be from the same account.

**Review by medyagh — CHANGES_REQUESTED** [review]
Thank you for this PR, do u mind clarifying how we can get the WINGET_TOKEN secret? I believe we need to add that to our github secrets

**Comment by vedantmgoyal2009** [comment]
> Thank you for this PR, do u mind clarifying how we can get the WINGET_TOKEN secret? I believe we need to add that to our github secrets

@medyagh thanks for getting back. you can create a new token by visiting https://github.com/settings/tokens/new.

**Comment by spowelljr** [comment]
> Can we fork winget-pkgs repo under @k8s-ci-robot or @minikube-bot account?
> 
> If yes, I can modify the workflow to use the fork from your personal account instead of organization account :)
> 
> Note: The GitHub Personal Access Token (PAT) and the fork should be from the same account.

Hi @vedantmgoyal2009, sorry for the delay. Yes, I can fork winget-pkgs as @minikube-bot, I'll do it tomorrow.

**Comment by spowelljr** [comment]
@vedantmgoyal2009 Repo is forked https://github.com/minikube-bot/winget-pkgs

**Comment by vedantmgoyal2009** [comment]
@spowelljr just one thing more... create a github personal access token (PAT) on @minikube-bot and add it as repository secret named `WINGET_TOKEN`.

**Comment by spowelljr** [comment]
> @spowelljr just one thing more... create a github personal access token (PAT) on @minikube-bot and add it as repository secret named `WINGET_TOKEN`.

What permissions does the PAT need?

**Comment by vedantmgoyal2009** [comment]
@spowelljr only one permission is needed: `public_repo`

**Comment by spowelljr** [comment]
> @spowelljr only one permission is needed: `public_repo`

Just created it

File: .github/workflows/winget.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Publish to WinGet
+on:
+  release:
+    types: [released]
+jobs:
+  publish:
+    runs-on: windows-latest # action can only be run on windows
+    steps:
+      - uses: vedantmgoyal2009/winget-releaser@v2
```
Comment (spowelljr) [inline .github/workflows/winget.yml:9]:
```suggestion
      - uses: vedantmgoyal2009/winget-releaser@79853c0938cc9946c1ec3cdd1b2e761bb0372b8c
```
For security reasons, we use the commit instead

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Comment by vedantmgoyal2009** [comment]
> > @spowelljr only one permission is needed: `public_repo`
> 
> Just created it and added it as a repo secret to https://github.com/minikube-bot/winget-pkgs, right?

Seems like I couldn't make myself clear the first time. The secret should be added to this repository so that this workflow can access it.

**Comment by vedantmgoyal2009** [comment]
> > @spowelljr only one permission is needed: `public_repo`
> 
> Just created it and added it as a repo secret to https://github.com/minikube-bot/winget-pkgs, right?

Seems like I couldn't make myself clear the first time. The secret should be added to this repository so that this workflow can access it.

**Comment by spowelljr** [comment]
Done

**Comment by vedantmgoyal2009** [comment]
We can merge now :)

**Review by spowelljr — APPROVED** [review]
Awesome, thanks for your patience!

**Pull request merged**